contracts/accountants/GenericAccountant.vy

```diff
@@ -50,59 +97,125 @@ MAX_SHARE: constant(uint256) = 7_500
-    self.asset = asset
+def __init__(
+    fee_manager: address, 
+    fee_recipient: address,
+    default_management: uint16, 
+    default_performance: uint16, 
+    default_refund: uint16, 
+    default_max: uint16
+):
+    """
+    @notice Initialilze the accountant and default fee config.
+    @param fee_manager Address to be in charge of this accountant.
+    @param default_management Default annual management fee to charge.
+    @param default_performance Default performance fee to charge.
+    @param default_refund Default refund ratio to give back on losses.
+    @param default_max Default max fee to allow as a perfent of gain.
+    """
+    assert fee_manager != empty(address), "ZERO ADDRESS"
+    assert fee_recipient != empty(address), "ZERO ADDRESS"
+    assert default_management <= self._management_fee_threshold(), "exceeds management fee threshold"
+    assert default_performance <= self._performance_fee_threshold(), "exceeds performance fee threshold"
+
+    # Set initial addresses
+    self.fee_manager = fee_manager
+    self.fee_recipient = fee_recipient
+
+    # Set the default fee config
+    self.default_config = Fee({
+        management_fee: default_management,
+        performance_fee: default_performance,
+        refund_ratio: default_refund,
+        max_fee: default_max
+    })
+
+    log UpdateDefaultFeeConfig(self.default_config)
 
 
 @external
 def report(strategy: address, gain: uint256, loss: uint256) -> (uint256, uint256):
-    """ """
-    total_refunds: uint256 = 0
-
-    # management_fee is charged in both profit and loss scenarios
+    """ 
+    @notice To be called by a vault during the process_report in which the accountant
+        will charge fees based on the gain or loss the strategy is reporting.
+    @dev Can only be called by a vault that has been added to this accountant.
+        Will default to the default_config for all amounts unless a custom config
+        has been set for a specific strategy.
+    @param strategy The strategy that is reporting.
+    @param gain The profit the strategy is reporting if any.
+    @param loss The loss the strategy is reporting if any.
+    """
+    # Make sure this is a valid vault.
+    assert self.vaults[msg.sender], "!authorized"
+
+    # Retrieve the strategies params from the vault.
     strategy_params: StrategyParams = IVault(msg.sender).strategies(strategy)
-    fee: Fee = self.fees[strategy]
-    duration: uint256 = block.timestamp - strategy_params.last_report
+    # Expected behavior is to use the default config.
+    fee: Fee = self.default_config
+
+    # Use custom fees if applicable.
+    if self.custom[strategy]:
+        fee = self.fees[strategy]
+
     total_fees: uint256 = 0
+    total_refunds: uint256 = 0
 
+    # Charge management fees no matter gain or loss.
     if fee.management_fee > 0:
-        #management_fee
+        # Time since last harvest.
+        duration: uint256 = block.timestamp - strategy_params.last_report
```

Comment:
its trusting the vault though not the strategy. Which it already has a built in trust assumption with given that its been whitelisted to the accountant.

Keeping an internal account of the reports would cause extra storage read and writes each time and mean management fees will not be charged correctly for the first report of a strategy on a specific accountant.

Though maybe thats actually a reasonable thing to do since its possible  for a strategy to be added for a while without any debt, so charging a management fee the first time on a strategy based on when it was added is probably an inaccurate way to do it as well

Reply:
So the current problem with manually tracking reports, is it would need to be done on another separate mapping.

Cause if we cant store the last_report if the fee struct when using the default config

I think given the expectation the vault is trusted from a code perspective, and if your adding it to the accountant you should trust management of it. Adding another double mapping to both read and write to every report seems to be not worth it.